operator/v1/types_dns.go

```diff
@@ -130,6 +131,61 @@ type Server struct {
 	ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
 }
 
+// DNSTransport indicates what type of connection should be used
+// +kubebuilder:validation:Enum=tls;cleartext
+type DNSTransport string
+
+const (
+	// TLSTransport indicates that TLS should be used for the connection
+	TLSTransport DNSTransport = "tls"
+
+	// CleartextTransport indicates that no encryption should be used for
+	// the connection.
+	CleartextTransport DNSTransport = "cleartext"
+)
+
+// DNSOverTLSConfig groups related configuration parameters used for configuring
+// forwarding to upstream resolvers that support DNS-over-TLS
+//
+type DNSOverTLSConfig struct {
+	// transport allows cluster administrators to opt-in to using a DNS-over-TLS
+	// connection between cluster DNS and an upstream resolver(s). Configuring
+	// TLS as the transport at this level without configuring a CABundle will
+	// result in the system certificates being used to verify the serving
+	// certificate of the upstream resolver(s).
+	//
+	// Possible values:
+	// "" (empty) - This is the default and results in DNS queries being sent over cleartext.
```

Comment:
Empty value means no opinion and the platform chooses the a default, which is subject to change over time.  The current default is cleartext.